### PR TITLE
SPUI-850/514/518 > Wallet screen UI

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

**Comment by eli1stark** [comment]
return back assets for fast_rsa and other functionality, reminder before review

File: lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart [hunk 1]
```diff
@@ -0,0 +1,748 @@
...
+      ).notifier,
+    );
+    final transactionHistory = useProvider(
+      operationHistoryNotipod(
+        widget.item.assetId,
+      ),
+    );
+    final screenHeight = MediaQuery.of(context).size.height;
+
+    var walletBackground = 'assets/images/green_wallet_gradient.svg';
```
Comment (eli1stark) [inline lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart:84]: move harcoded asset names to constants in shared folder, fix all occurences

File: lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart [hunk 2]
```diff
@@ -0,0 +1,748 @@
...
+
+    if (!widget.item.isGrowing) {
+      walletBackground = 'assets/images/red_wallet_gradient.svg';
```
Comment (eli1stark) [inline lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart:87]: move harcoded asset names to constants in shared folder, fix all occurences

File: lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart [hunk 3]
```diff
@@ -0,0 +1,748 @@
...
+import '../../../helper/format_date.dart';
+import '../../../notifier/operation_history_notipod.dart';
+import '../../../notifier/operation_history_union.dart';
+import '../../../provider/operation_history_fpod.dart';
+import 'components/card_block/components/wallet_card.dart';
+import 'components/card_block/components/wallet_card_collapsed.dart';
+import 'components/transactions_list_item/transaction_list_item.dart';
+import 'components/transactions_list_item/transaction_list_loading_item.dart';
+
+class WalletBody extends StatefulHookWidget {
```
Comment (eli1stark) [inline lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart:21]: add todo: refactor this widget

File: lib/shared/components/app_builder.dart [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import '../helpers/device_size_from.dart';
+import '../providers/device_size/device_size_pod.dart';
+
+class AppBuilder extends HookWidget {
```
Comment (eli1stark) [inline lib/shared/components/app_builder.dart:8]: remove this widget, this is already implemented in the core, remember?, looks like your old changes want to override the new ones

File: plugins/simple_kit/lib/src/notifications/view/show_simple_notification.dart [hunk 5]
```diff
@@ -8,22 +8,26 @@ void showSNotification({
...
-        backgroundColor: Colors.transparent,
-        behavior: FlashBehavior.fixed,
-        position: FlashPosition.top,
-        useSafeArea: false,
-        child: SPaddingH24(
-          child: SNotificationBox(
-            text: text,
-          ),
-        ),
+  WidgetsBinding.instance?.addPostFrameCallback(
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/notifications/view/show_simple_notification.dart:11]: why you did this?

File: plugins/simple_kit/lib/src/shared/components/simple_skeleton_text_loader.dart [hunk 6]
```diff
@@ -6,7 +6,14 @@ import '../../colors/view/simple_colors_light.dart';
 final _loaderRadius = BorderRadius.circular(20.0);
 
 class SSkeletonTextLoader extends StatefulHookWidget {
-  const SSkeletonTextLoader({Key? key}) : super(key: key);
+  const SSkeletonTextLoader({
+    Key? key,
+    this.height = 16,
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/components/simple_skeleton_text_loader.dart:11]: make this parameter required to make this component more general without default size

File: plugins/simple_kit/lib/src/shared/components/simple_skeleton_text_loader.dart [hunk 7]
```diff
@@ -6,7 +6,14 @@ import '../../colors/view/simple_colors_light.dart';
...
+    this.width = 80,
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/shared/components/simple_skeleton_text_loader.dart:12]: make this parameter required to make this component more general without default size

File: plugins/simple_kit/lib/src/buttons/basic_buttons/base/simple_base_button.dart [hunk 8]
```diff
@@ -9,6 +9,7 @@ class SimpleBaseButton extends StatelessWidget {
     Key? key,
     this.icon,
     this.onTap,
+    this.addIconPadding = true,
```
Comment (eli1stark) [inline plugins/simple_kit/lib/src/buttons/basic_buttons/base/simple_base_button.dart:12]: rename to enableIconPadding

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: assets/images/deposit_icon.svg [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply assets/images/deposit_icon.svg:1]: done

File: assets/images/minus_icon.svg [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply assets/images/minus_icon.svg:1]: done

File: assets/images/paid_interest_rate_icon.svg [hunk 11]
```diff
@@ -0,0 +1,5 @@
+<svg width="18" height="18" viewBox="0 0 18 18" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply assets/images/paid_interest_rate_icon.svg:1]: done

File: assets/images/plus_icon.svg [hunk 12]
```diff
@@ -0,0 +1,4 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply assets/images/plus_icon.svg:1]: done

File: assets/images/receive_by_phone_icon.svg [hunk 13]
```diff
@@ -0,0 +1,5 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply assets/images/receive_by_phone_icon.svg:1]: done

File: assets/images/reward_payment_icon.svg [hunk 14]
```diff
@@ -0,0 +1,6 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply assets/images/reward_payment_icon.svg:1]: done

File: assets/images/send_by_phone_icon.svg [hunk 15]
```diff
@@ -0,0 +1,5 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply assets/images/send_by_phone_icon.svg:1]: done

File: assets/images/send_icon.svg [hunk 16]
```diff
@@ -0,0 +1,5 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply assets/images/send_icon.svg:1]: done

File: assets/images/swap_icon.svg [hunk 17]
```diff
@@ -0,0 +1,7 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply assets/images/swap_icon.svg:1]: done

File: assets/images/withdrawal_fee_icon.svg [hunk 18]
```diff
@@ -0,0 +1,5 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply assets/images/withdrawal_fee_icon.svg:1]: done

File: lib/app/shared/features/wallet/view/components/action_button/action_button.dart [hunk 19]
```diff
@@ -0,0 +1,326 @@
...
+                  ? const Cubic(0.42, 0, 0, 0.99)
+                  : const Cubic(1, 0, 0.58, 1),
+              child: Stack(
+                children: [
+                  AnimatedOpacity(
+                    opacity: actionActive.value ? 1 : 0,
+                    duration:
+                        Duration(milliseconds: actionActive.value ? 150 : 300),
+                    curve: actionActive.value
+                        ? Interval(
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/action_button/action_button.dart:96]: done

File: lib/app/shared/features/wallet/view/components/action_button/action_button.dart [hunk 20]
```diff
@@ -0,0 +1,326 @@
...
+                            0.0,
+                            0.5,
+                            curve: actionActive.value
+                                ? const Cubic(0.42, 0, 0, 0.99)
+                                : const Cubic(1, 0, 0.58, 1),
+                          )
+                        : Interval(
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/action_button/action_button.dart:103]: done

File: lib/app/shared/features/wallet/view/components/action_button/action_button.dart [hunk 21]
```diff
@@ -0,0 +1,326 @@
...
+                        updateActionState();
+                      },
+                    ),
+                  ),
+                  AnimatedOpacity(
+                    opacity: actionActive.value ? 0 : 1,
+                    duration:
+                        Duration(milliseconds: actionActive.value ? 300 : 150),
+                    curve: actionActive.value
+                        ? Interval(
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/action_button/action_button.dart:211]: done

File: lib/app/shared/features/wallet/view/components/action_button/action_button.dart [hunk 22]
```diff
@@ -0,0 +1,326 @@
...
+                            0.0,
+                            0.5,
+                            curve: actionActive.value
+                                ? const Cubic(0.42, 0, 0, 0.99)
+                                : const Cubic(1, 0, 0.58, 1),
+                          )
+                        : Interval(
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/action_button/action_button.dart:218]: done

File: lib/app/shared/features/wallet/view/components/wallet_body/components/card_block/components/wallet_card.dart [hunk 23]
```diff
@@ -0,0 +1,131 @@
...
+              margin: const EdgeInsets.only(top: 32),
+              decoration: BoxDecoration(
+                borderRadius: BorderRadius.circular(20),
+                color: colors.green,
+              ),
+              child: SBaselineChild(
+                baseline: 17,
+                child: Text(
+                  '+\$120.23',
+                  style: sSubtitle3Style.copyWith(color: colors.white),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallet_body/components/card_block/components/wallet_card.dart:75]: done

File: lib/app/shared/features/wallet/view/components/wallet_body/components/card_block/components/wallet_card.dart [hunk 24]
```diff
@@ -0,0 +1,131 @@
...
+            child: SBaselineChild(
+              baseline: 24,
+              child: Text(
+                formatCurrencyAmount(
+                  symbol: currency.assetId,
+                  value: currency.assetBalance,
+                  accuracy: currency.accuracy,
+                  prefix: currency.prefixSymbol,
+                ),
+                style: sBodyText2Style.copyWith(color: colors.grey1),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallet_body/components/card_block/components/wallet_card.dart:112]: done

File: lib/app/shared/features/wallet/model/transaction_icon_model.dart [hunk 25]
```diff
@@ -0,0 +1,11 @@
+import 'package:flutter/material.dart';
+
+class TransactionIconModel {
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/model/transaction_icon_model.dart:3]: done

File: lib/app/shared/features/wallet/view/components/wallet_body/components/card_block/components/wallet_card_collapsed.dart [hunk 26]
```diff
@@ -0,0 +1,69 @@
...
+          SBaselineChild(
+            baseline: 20,
+            child: Text(
+              formatCurrencyAmount(
+                symbol: currency.assetId,
+                value: currency.assetBalance,
+                accuracy: currency.accuracy,
+                prefix: currency.prefixSymbol,
+              ),
+              style: sBodyText2Style.copyWith(color: colors.grey1),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallet_body/components/card_block/components/wallet_card_collapsed.dart:60]: done

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/components/common_transaction_details_block.dart [hunk 27]
```diff
@@ -21,40 +19,32 @@ class CommonTransactionDetailsBlock extends HookWidget {
...
-          style: TextStyle(
-            fontSize: 40.sp,
-            fontWeight: FontWeight.w600,
+          style: sTextH1Style,
+        ),
+        Text(
+          DateFormat('EEEE, MMMM d, y').format(
+            DateTime.parse('${transactionListItem.timeStamp}Z').toLocal(),
           ),
+          style: sBodyText2Style.copyWith(color: colors.grey2),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/components/common_transaction_details_block.dart:45]: done

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/receive_details.dart [hunk 28]
```diff
@@ -0,0 +1,57 @@
...
+              children: [
+                TransactionDetailsValueText(
+                  text:
+                      '${transactionListItem.receiveByPhoneInfo!
+                          .fromPhoneNumber}',
+                ),
+                if (senderName.isNotEmpty) ...[
+                  Text(
+                    senderName,
+                    style: sBodyText2Style.copyWith(color: colors.grey1),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/components/transaction_details/receive_details.dart:47]: done

File: lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/transaction_list_item.dart [hunk 29]
```diff
@@ -1,124 +1,124 @@
...
-  String _sellAmount(bool hidden) =>
-      'For ${'${hidden ? '???' : transactionListItem.swapInfo!.buyAmount}'} '
-      '${transactionListItem.swapInfo!.buyAssetId}';
-
-  String _balanceChange(bool hidden) =>
-      '${hidden ? '???' : transactionListItem.balanceChange} '
-      '${transactionListItem.assetId}';
-
-  IconData _icon(OperationType type) {
+  String _icon(OperationType type) {
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallet_body/components/transactions_list_item/transaction_list_item.dart:96]: refactored to usage of simple_kit_icons

File: lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart (hunk 1 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart:84]: done

File: lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart (hunk 2 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart:87]: done

File: lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart (hunk 3 above)
Comment (VovaMedvediev) [reply lib/app/shared/features/wallet/view/components/wallet_body/wallet_body.dart:21]: done

File: lib/shared/components/app_builder.dart (hunk 4 above)
Comment (VovaMedvediev) [reply lib/shared/components/app_builder.dart:8]: removed

File: plugins/simple_kit/lib/src/buttons/basic_buttons/base/simple_base_button.dart (hunk 8 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/buttons/basic_buttons/base/simple_base_button.dart:12]: done

File: plugins/simple_kit/lib/src/shared/components/simple_skeleton_text_loader.dart (hunk 7 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/shared/components/simple_skeleton_text_loader.dart:12]: done

File: plugins/simple_kit/lib/src/shared/components/simple_skeleton_text_loader.dart (hunk 6 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/shared/components/simple_skeleton_text_loader.dart:11]: done

File: plugins/simple_kit/lib/src/notifications/view/show_simple_notification.dart (hunk 5 above)
Comment (VovaMedvediev) [reply plugins/simple_kit/lib/src/notifications/view/show_simple_notification.dart:11]: It wasn't working at Wallet Screen without addPostFrameCallback. But I tested it and it is working. Deleted

File: assets/images/paid_interest_rate_icon.svg (hunk 11 above)
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply assets/images/paid_interest_rate_icon.svg]: done

**Pull request merged**